txircd/ircd.py

```diff
@@ -385,7 +375,150 @@ def reloadModule(self, moduleName):
 		else:
 			deferList.addCallback(lambda result: self.loadModule(moduleName))
 		return deferList
-	
+
+	def verifyConfig(self, config):
+		# Verify the configuration values that are used in the core first.
+
+		# IRCd
+		if "server_name" not in config:
+			raise ConfigValidationError("server_name", "required item not found in configuration file.")
+		if not isinstance(config["server_name"], basestring):
+			raise ConfigValidationError("server_name", "value must be a string")
+		if len(config["server_name"]) > 64:
+			config["server_name"] = config["server_name"][:64]
+			self.logConfigValidationWarning("server_name", "value is too long", config["server_name"])
+		if not re.match(r"^[a-zA-Z0-9.-]+\.[a-zA-Z0-9.-]+$", config["server_name"]):
+			raise ConfigValidationError("server_name", "server name must look like a valid hostname.")
+		if "server_id" in config:
+			if not isinstance(config["server_id"], basestring):
+				raise ConfigValidationError("server_id", "value must be a string")
+			else:
+				config["server_id"] = config["server_id"].upper()
+		else:
+			randFromName = random.Random(config["server_name"])
+			config["server_id"] = randFromName.choice(string.digits)
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
+			config["server_id"] += randFromName.choice(string.digits + string.uppercase)
+		if len(config["server_id"]) != 3 or not config["server_id"].isalnum() or not config["server_id"][0].isdigit():
+			raise ConfigValidationError("server_id", "value must be a 3-character alphanumeric string starting with a number.")
+		if "server_description" not in config:
+			raise ConfigValidationError("server_description", "required item not found in configuration file.")
+		if not isinstance(config["server_description"], basestring):
+			raise ConfigValidationError("server_description", "value must be a string")
+		if "network_name" not in config:
+			raise ConfigValidationError("network_name", "required item not found in configuration file.")
+		if not isinstance(config["network_name"], basestring):
+			raise ConfigValidationError("network_name", "value must be a string")
+		if len(config["network_name"]) > 32:
+			config["network_name"] = config["network_name"][:32]
+			self.logConfigValidationWarning("network_name", "value is too long", config["network_name"])
+		if not "bind_client" not in config:
+			config["bind_client"] = [ "tcp:6667:interface={::}" ]
+			self.logConfigValidationWarning("bind_client", "no default client binding specified", "[ \"tcp:6667:interface={::}\" ]")
```

Comment:
I think it should be a warning. In my opinion this default shouldn't exist in the first place or be bound to localhost only. But that's just me.